onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -140,6 +140,17 @@ namespace Dml::GraphDescBuilder
         {
             intermediateEdge.FromNodeIndex = shiftedIndicesMapping[intermediateEdge.FromNodeIndex];
             intermediateEdge.ToNodeIndex = shiftedIndicesMapping[intermediateEdge.ToNodeIndex];
+            // We need to update the edge name only when the name contains the intermediateEdge.FromNodeIndex
+            std::string oldEdgeName = intermediateEdge.Name;
+            size_t pos = oldEdgeName.find("nodeIdx:");
+            if (pos != std::string::npos)
+            {
+                if (pos != 0)
+                {
+                    intermediateEdge.Name = oldEdgeName.substr(0, pos);
+                }
+                intermediateEdge.Name += "nodeIdx:" + std::to_string(intermediateEdge.FromNodeIndex) + "-outputIdx:" + std::to_string(intermediateEdge.FromNodeOutputIndex);
+            }
```

Comment:
Is this needed when not doing serialization? I would put it behind an ifdef otherwise.

Reply:
We don't use ifdef for serialization anymore, it is handled by an EP option now. We can still gate it by wrapping it inside an `if` block based on the `graphSerializationEnabled` flag but I think we should not do that. It is so because not updating the edge name might not impact the end-to-end flow if serialization is not enabled, but the edge names will remain inconsistent. 

Also, in future if we use the edge names anywhere else apart from serialization, then it might cause an issue.